### PR TITLE
op-chain-ops: check-ecotone improvements

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

- 4788/4844 mixup fix (the 4788 contract test was named as the 4844 precompile test, and the precompile check was missing)
- new 4844 precompile test
- fix edge-case of 4788 contract test that happens during checks around deployment: if the contract is new (i.e. deployed with the upgrade tx rather than someone doing the 4788 tx manually) in the ecotone upgrade block, then the 4788 root of the first ecotone block is not processed yet.
- log starting balance
- more logging of progress

### CODE REVIEW

**Review by trianglesphere — APPROVED** [review]

File: op-chain-ops/cmd/check-ecotone/main.go [hunk 1]
```diff
@@ -262,12 +340,25 @@ func check4844Precompile(ctx context.Context, env *actionEnv) error {
...
+		// (the beacon block root is processed at the start of the block,
+		// but the contract might not exist yet, during activation).
+		revert := ti%conf.BlockTime != alignment
+		if conf.IsEcotoneActivationBlock(ti) {
+			// if the contract already existed, then we deployed the nonce=1 contract during upgrade
+			code, err := env.l2.CodeAt(ctx, crypto.CreateAddress(derive.EIP4788From, 1), nil)
+			if err != nil {
+				return fmt.Errorf("failed to check code: %w", err)
+			}
+			revert = revert || len(code) > 0
```
Comment (protolambda) [inline op-chain-ops/cmd/check-ecotone/main.go]:
```suggestion
			revert = revert || len(code) == 0
```

Correction: when the nonce=1 contract doesn't exist, then the nonce=0 didn't exist during upgrade, and thus reverts.

This isn't covered when already way past the upgrade, but happens if running the check at or closely after the upgrade happens.

**Review by sebastianst — COMMENTED** [review]
LGTM but would like to get https://github.com/ethereum-optimism/optimism/pull/9701 merged into this

**Comment by protolambda** [comment]
Merged in suggestions from @sebastianst and rebased on develop to get the CI fix in.

**Comment by trianglesphere** [comment]
Failing `go-mod-download`, kinda looks like a lint issue

**Review by sebastianst — APPROVED** [review]
LGTM now, ran latest version locally against OP `sepolia-devnet-0` ✅

**Pull request merged**